### PR TITLE
WIP: Allow monitoring_role_arn to be specified

### PR DESCRIPTION
**Pull request by robinbowes**
if `monitoring_interval` is set to a non-zero value then `monitoring_role_arn` must also be set.

### CODE REVIEW

File: variables.tf [hunk 1]
```diff
@@ -146,3 +146,12 @@ variable "monitoring_interval" {
   description = "To disable collecting Enhanced Monitoring metrics, specify 0. The default is 0. Valid Values: 0, 1, 5, 10, 15, 30, 60."
   default     = "0"
 }
+
+variable "monitoring_role_arn" {
+  description = <<-EOF
```
Comment (antonbabenko) [inline variables.tf]: Let's use double quotes strings, as we do in all other places for consistency.

**Comment by antonbabenko** [comment]
New release - https://github.com/terraform-community-modules/tf_aws_rds/releases/tag/v1.0.4

**Comment by robinbowes** [comment]
Sorry - can you revert this? I was experimenting with some `depends_on` changes and broke this branch.

**Comment by robinbowes** [comment]
```
depends on value cannot contain interpolations: ${var.depends_on}
```

**Comment by antonbabenko** [comment]
Reverted.

**Comment by robinbowes** [comment]
The problem I have is that I need to make sure the role that is used for the monitoring_role_arn is created before the `aws_db_instance`.

**Comment by antonbabenko** [comment]
I see what you mean. A couple of bad solutions come to my mind.

The first workaround is to create monitoring role in this module as well, but it is a bad idea because users may want to reuse existing monitoring role.

Another bad solution is to use use something like `provisioner "local-exec" {  command = "sleep 30" }`.

Sorry, I am out of good solutions on Fridays :)

**Comment by robinbowes** [comment]
The current state of terraform is reminding me a lot of the early days of puppet, ie. clearly a fantastic tool but missing a few important implementation details.

**Pull request merged**